coderd/notifications/notifier.go

```diff
@@ -228,22 +228,10 @@ func (n *notifier) prepare(ctx context.Context, msg database.AcquireNotification
 		return nil, xerrors.Errorf("failed to resolve handler %q", msg.Method)
 	}
 
-	helpers := make(template.FuncMap)
-	for k, v := range n.helpers {
-		helpers[k] = v
-	}
-
-	appName, err := n.fetchAppName(ctx)
+	helpers, err := n.fetchHelpers(ctx)
 	if err != nil {
-		return nil, xerrors.Errorf("fetch app name: %w", err)
+		return nil, xerrors.Errorf("fetch helpers: %w", err)
```

Comment:
Have you validated that failures here cause a "temporary failure" so that subsequent requests are retried?